astronomer/providers/google/cloud/triggers/kubernetes_engine.py

```diff
@@ -0,0 +1,154 @@
+        *,
+        location: str,
+        cluster_name: str = "default",
+        use_internal_ip: bool = False,
+        project_id: Optional[str] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        is_delete_operator_pod: bool = True,
+        regional: bool = False,
+        namespace: Optional[str] = None,
+        name: Optional[str] = None,
+        cluster_context: Optional[str] = None,
+        in_cluster: Optional[bool] = None,
+        poll_interval: float = 5.0,
+        pending_phase_timeout: float = 120.0,
+    ):
+        super().__init__()
+        self.location = location
+        self.cluster_name = cluster_name
+        self.regional = regional
+        self.use_internal_ip = use_internal_ip
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.cluster_context = cluster_context
+        self.in_cluster = in_cluster
+        self.namespace = namespace
+        self.is_delete_operator_pod = is_delete_operator_pod
+        self.name = name
+        self.poll_interval = poll_interval
+        self.pending_phase_timeout = pending_phase_timeout
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serialize GKEStartPodTrigger object"""
+        return (
+            "astronomer.providers.google.cloud.triggers.kubernetes_engine.GKEStartPodTrigger",
+            {
+                "project_id": self.project_id,
+                "cluster_name": self.cluster_name,
+                "impersonation_chain": self.impersonation_chain,
+                "regional": self.regional,
+                "location": self.location,
+                "use_internal_ip": self.use_internal_ip,
+                "gcp_conn_id": self.gcp_conn_id,
+                "is_delete_operator_pod": self.is_delete_operator_pod,
+                "name": self.name,
+                "namespace": self.namespace,
+                "poll_interval": self.poll_interval,
+                "pending_phase_timeout": self.pending_phase_timeout,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Wait for pod to reach terminal state"""
+        with GKEStartPodOperator.get_gke_config_file(
+            gcp_conn_id=self.gcp_conn_id,
+            project_id=self.project_id,
+            cluster_name=self.cluster_name,
+            impersonation_chain=self.impersonation_chain,
+            regional=self.regional,
+            location=self.location,
+            use_internal_ip=self.use_internal_ip,
+        ) as config_file:
+            hook_params: Optional[Dict[str, Any]] = {
+                "cluster_context": self.cluster_context,
+                "config_file": config_file,
+                "in_cluster": self.in_cluster,
+            }
+            hook = KubernetesHookAsync(conn_id=None, **(hook_params or {}))
```

Comment:
```suggestion
            hook = KubernetesHookAsync(conn_id=None, **hook_params)
```
looks like hook_params will always be a dict with the code as you have it now

Reply:
yes